safenode/src/node/spendbook.rs

```diff
@@ -0,0 +1,352 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+use crate::domain::dbc_genesis::{is_genesis_parent_tx, GENESIS_DBC};
+use crate::network::{close_group_majority, Network};
+use crate::protocol::error::{Error, Result};
+use crate::protocol::messages::{Query, QueryResponse, Request, Response};
+use crate::protocol::storage::DbcAddress;
+
+use std::collections::{BTreeMap, BTreeSet};
+
+use libp2p::kad::{Record, RecordKey};
+use sn_dbc::{DbcTransaction, SignedSpend};
+use tokio::sync::RwLock;
+
+/// The entitiy managing spends in a Node
+#[derive(Default)]
+pub(crate) struct SpendBook {
```

Comment:
I think `SpendBook` is not the right level of abstraction here.

I was talking with Dirvine about this recently, and we agreed that the `SpendBook` is _the entire network_.

A `spend` is like a single entry (word) on one page of that book.
A single node holds a page out of the book. 

But a node doesn't hold the book.

Reply:
Interesting idea, what should we rename it to then?